src/sentry/utils/dates.py

```diff
@@ -115,3 +116,17 @@ def get_rollup_from_request(request, params, default_interval, error, top_events
     if date_range.total_seconds() / interval.total_seconds() > max_rollup_points:
         raise error
     return int(interval.total_seconds())
+
+
+def outside_retention(start, end, organization):
+    """
+    Check if a start-end datetime range falls is outside an
+    organizations retention period.
+    """
+    retention = quotas.get_event_retention(organization=organization)
+    if not retention:
+        return False
+
+    now = datetime.utcnow().astimezone(pytz.utc) if start.tzinfo else datetime.utcnow()
```

Comment:
The start/end datetimes from `src/sentry/utils/snuba.py` are naive, so I'm handling it here.

Reply:
Ah I know why it's failing